modules/system/defaults/CustomPreferences.nix

```diff
@@ -2,10 +2,14 @@
 
 with lib;
 
-{
+let
+  defaultsType = types.submodule {
+    freeformType = types.anything;
```

Comment:
Is there a better type than `types.anything` here? Maybe something related to `plist`?

Reply:
I added a stricter type check that we could add to nixpkgs as `lib.formats.plist` later.